### PR TITLE
CLN: D213: Multi-line docstring summary should start at the second line

### PR DESCRIPTION
**Pull request by simonjayhawkins**

### CODE REVIEW

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -819,7 +819,7 @@ def tz_convert(self, tz):
         dtype = tz_to_dtype(tz)
         return self._simple_new(self.asi8, dtype=dtype, freq=self.freq)
 
-    def tz_localize(self, tz, ambiguous="raise", nonexistent="raise"):
+    def tz_localize(self, tz, ambiguous="raise", nonexistent="raise"):  # noqa
```
Comment (jbrockmendel) [inline pandas/core/arrays/datetimes.py]: is this related to the rest of the PR?

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -86,7 +86,8 @@ class IndexingError(Exception):
 
 
 class IndexingMixin:
-    """Mixin for adding .loc/.iloc/.at/.iat to Datafames and Series.
+    """
+    Mixin for adding .loc/.iloc/.at/.iat to Datafames and Series.
     """
```
Comment (jbrockmendel) [inline pandas/core/indexing.py]: do we have a policy on whether this should be one line or not?

File: pandas/io/common.py [hunk 3]
```diff
@@ -74,7 +74,8 @@ def is_url(url) -> bool:
 def _expand_user(
     filepath_or_buffer: FilePathOrBuffer[AnyStr],
 ) -> FilePathOrBuffer[AnyStr]:
-    """Return the argument with an initial component of ~ or ~user
+    """
+    Return the argument with an initial component of ~ or ~user
        replaced by that user's home directory.
```
Comment (jbrockmendel) [inline pandas/io/common.py]: this introduces a weird alignment here

**Comment by jbrockmendel** [comment]
Mostly looks fine, only question is if we want to avoid introducing alignment mismatches

File: pandas/core/arrays/datetimes.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/core/arrays/datetimes.py]: that shouldn't have been left in, the pydocstyle cli is crashing on that docstring. will need to raise an issue(look into this further), if we want to use this tool for cli.

will remove.

File: pandas/core/indexing.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/indexing.py]: there is `D200 One-line docstring should fit on one line with quotes`, see http://www.pydocstyle.org/en/5.0.2/error_codes.html, but I think that would go against the code cleanups in #31162, #31462 and others.

we want consistency. so we need a policy.

File: pandas/io/common.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/io/common.py]: this PR is just D213, the alignment will be fixed with D207/D208

**Comment by WillAyd** [comment]
Can you merge master?

File: pandas/io/common.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/io/common.py]: so some overlap here with #31890, and D207/D208 will need to be rerun after D209/D213 merged

File: pandas/io/excel/_base.py [hunk 4]
```diff
@@ -755,7 +756,8 @@ def _value_with_fmt(self, val):
 
     @classmethod
     def check_extension(cls, ext):
-        """checks that path's extension against the Writer's supported
+        """
+        checks that path's extension against the Writer's supported
         extensions.  If it isn't supported, raises UnsupportedFiletypeError."""
```
Comment (jorisvandenbossche) [inline pandas/io/excel/_base.py]: Since you are changing things here, can you put the ending triple quotes on their own line as well? (other cases below as well)

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: pandas/io/excel/_base.py (hunk 4 above)
Comment (simonjayhawkins) [reply pandas/io/excel/_base.py]: see #31891

Comment (simonjayhawkins) [reply pandas/io/excel/_base.py]: probably best to merge #31891 first, since less violations there and once merged this diff should look good and any leftover misalignment can be pushed here.

File: pandas/core/indexing.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/indexing.py]: there are ~550 one-line docstrings split over 3 lines.

File: pandas/io/common.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/io/common.py]: i'll push the fix in this PR. (prob thur)

File: pandas/core/generic.py [hunk 5]
```diff
@@ -3120,18 +3125,21 @@ def to_csv(
...
+        """
+        Set the _cacher attribute on the calling object with a weakref to
         cacher.
         """
         self._cacher = (item, weakref.ref(cacher))
 
     def _reset_cacher(self) -> None:
-        """Reset the cacher."""
+        """
+        Reset the cacher."""
```
Comment (simonjayhawkins) [inline pandas/core/generic.py]: i'll fix in this PR

Comment (simonjayhawkins) [reply pandas/core/generic.py]: done

File: pandas/io/excel/_base.py (hunk 4 above)
Comment (simonjayhawkins) [reply pandas/io/excel/_base.py]: done

File: pandas/io/common.py (hunk 3 above)
Comment (simonjayhawkins) [reply pandas/io/common.py]: done

**Review by jorisvandenbossche — APPROVED** [review]
